### PR TITLE
Bag of characters kernel

### PR DESCRIPTION
**Pull request by henrymoss**
I have added in support for the SMILES string kernel (https://www.tandfonline.com/doi/abs/10.1080/1062936X.2011.645874).

It proved immensely difficult to pass strings around to the kernels without gpytorch getting very unhappy. It turned out to be much easier to extract the required info from the strings through the datascaler rather than by building a new specific string kernel, i.e. I transform strings to a bag-of-characters representation using either of the new `featurizer("bag_of_smiles")` or `featurizer("bag_of_selfies")`  which can then be passed to any of the bit-vector kernels. 

I have also added support for SELFIE strings in the datascalar, allowing a bag-of-character representation of SELFIES to go alongside the SMILES one.

Finally, I have update the molecular regression notebook to demonstrate this new functionality.

Unfortunately, I accidental ran black which reformatted a lot of the files. The actual changes from this PR are only in mol_prop.py and the notebook tutorial

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ We recommend using a conda virtual environment.
 
 ```
 conda create -n gprotorch python==3.8
-pip install gpytorch botorch
+pip install gpytorch botorch selfies
 pip install scikit-learn pandas pytest tqdm
 conda install -c conda-forge rdkit
```
Comment (Ryan-Rhys) [inline README.md:12]: Looks like the notebook below was changed, was that on purpose?

File: gprotorch/dataloader/mol_prop.py [hunk 2]
```diff
@@ -73,7 +79,12 @@ def featurize(self, representation, bond_radius=3, nBits=2048):
         def fingerprints():
 
             rdkit_mols = [MolFromSmiles(smiles) for smiles in self.features]
-            fps = [AllChem.GetMorganFingerprintAsBitVect(mol, bond_radius, nBits=nBits) for mol in rdkit_mols]
+            fps = [
```
Comment (Ryan-Rhys) [inline gprotorch/dataloader/mol_prop.py:82]: This is Black's suggestion with the formatting I'm guessing?

File: README.md (hunk 1 above)
Comment (henrymoss) [reply README.md:12]: what do you mean?

File: gprotorch/dataloader/mol_prop.py (hunk 2 above)
Comment (henrymoss) [reply gprotorch/dataloader/mol_prop.py:82]: yh

**Pull request merged**